src/join/composer.jl

```diff
@@ -346,12 +399,33 @@ function _compose_joined_table(joiner::DataFrameJoiner, kind::Symbol, makeunique
         end
     end
 
+    was_sorted = false
+    if order == :left && !(issorted(left_ixs) && isempty(leftonly_ixs))
+        left_cols_idxs = _sort_compose_helper(nrow(joiner.dfl) + 1,
+                                              1:nrow(joiner.dfl), target_nrow,
+                                              left_ixs, lil + 1, leftonly_ixs, loil)
+        new_order = _count_sortperm(left_cols_idxs)
+        was_sorted = true
```

Comment:
Why is this called `was_sorted`, given that it determines whether we permute rows below so that they are sorted?

Reply:
OK. I removed it. Instead I initialize `new_order` to `nothing` and reorder things if `new_order` is not `nothing`.